flogger/flogger-api-tests/src/test/kotlin/com/google/common/flogger/LogContextSpec.kt

```diff
@@ -0,0 +1,999 @@
+        private val ONCE_PER_SECOND = newRateLimitPeriod(1, SECONDS)
+
+        // In normal use, the logger would never need to be passed in,
+        // and you'd use `logVarargs()`.
+        private fun logHelper(logger: FluentLogger2, logSite: LogSite, n: Int, message: String) {
+            logger.atInfo()
+                .withInjectedLogSite(logSite)
+                .every(n)
+                .log("%s", message)
+        }
+    }
+
+    @Test
+    fun `return 'true' if logging is enabled at the implied level`() {
+        backend.setLevel(INFO)
+        logger.atFine().isEnabled().shouldBeFalse()
+        logger.atInfo().isEnabled().shouldBeTrue()
+        logger.at(WARNING).isEnabled().shouldBeTrue()
+    }
+
+    @Test
+    fun `log with the given cause`() {
+        val cause = Throwable()
+        logger.atInfo()
+            .withCause(cause)
+            .log("Hello World")
+
+        backend.lastLogged.metadata shouldHaveSize 1
+        backend.lastLogged.metadata.shouldUniquelyContain(Key.LOG_CAUSE, cause)
+        backend.lastLogged shouldHaveMessage "Hello World"
+    }
+
+    @Test
+    fun `lazily evaluate arguments`() {
+        logger.atInfo().log("Hello %s", lazy { "World" })
+        logger.atFine().log(
+            "Hello %s",
+            lazy { error("Lazy arguments should not be evaluated in a disabled log statement") }
+        )
+
+        backend.lastLogged shouldHaveMessage "Hello %s"
+        backend.lastLogged.shouldHaveArguments("World")
+    }
+
+    @Test
+    fun `accept a formatted message`() {
+        logger.at(INFO).log("Formatted %s", "Message")
+        backend.loggedCount shouldBe 1
+        backend.lastLogged shouldHaveMessage "Formatted %s"
+        backend.lastLogged.shouldHaveArguments("Message")
+
+        // Should NOT ask for literal argument as none exists.
+        shouldThrow<IllegalStateException> {
+            backend.lastLogged.literalArgument
+        }
+    }
+
+    @Test
+    fun `accept a literal message`() {
+        logger.at(INFO).log("Literal Message")
+        backend.loggedCount shouldBe 1
+        backend.lastLogged shouldHaveMessage "Literal Message"
+
+        // Cannot ask for format arguments as none exist.
+        backend.lastLogged.templateContext.shouldBeNull()
+        shouldThrow<IllegalStateException> {
+            backend.lastLogged.getArguments()
+        }
+    }
+
+    @Test
+    fun `accept multiple metadata entries`() {
+        val cause = RuntimeException()
+        logger.atInfo()
+            .withCause(cause)
+            .every(42)
+            .log("Hello World")
+        backend.loggedCount shouldBe 1
+        backend.lastLogged.metadata shouldHaveSize 2
+        backend.lastLogged.metadata.shouldUniquelyContain(Key.LOG_EVERY_N, 42)
+        backend.lastLogged.metadata.shouldUniquelyContain(Key.LOG_CAUSE, cause)
+    }
+
+    @Test
+    fun `handle multiple metadata keys`() {
```

Comment:
In this test, I would make it less hard-coded, meaning the expected values should rely more on input, rather than just on human properly copying the values from the top of the test.

Other tests in this file would also benefit from such an improvement (please consider!). But this one just stands out, as the most fragile out of all.